code/src/reducers/tasks.js

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const tasks = createSlice({
+
+ 
+    name: "tasks",
+    initialState: {
+    items: [],
+     
+   
+  },
+ 
+    reducers: {
+      addTask:(state, action) =>{
+        const { text } = action.payload
+        state.items.push({
```

Comment:
To avoid mutation of the initial state we should not use the push-method, I think. You could instead try to create a new variable array, spread the initial state and then add the new task. I found this article and thought it was pretty good at explaining it.. 😊 (https://medium.com/javascript-in-plain-english/how-to-add-to-an-array-in-react-state-3d08ddb2e1dc) and this one was really good, too (https://medium.com/@thejasonfile/using-the-spread-operator-in-react-setstate-c8a14fc51be1) 📕🤓